src/libs/ascent/runtimes/expressions/ascent_conduit_reductions.cpp

```diff
@@ -481,6 +594,75 @@ struct SumFunctor
   }
 };
 
+struct DFAddFunctor
+{
+  template<typename T, typename Exec>
+  conduit::Node operator()(const MemoryAccessor<T> l_accessor,
+  			   const MemoryAccessor<T> r_accessor,
+                           const Exec &) const
+  {
+    const int l_size = l_accessor.m_size;
+    const int r_size = r_accessor.m_size;
+    bool diff_sizes = false;
+    int size; 
+    int max_size;
+
+    size = max_size = l_size; 
+    if(l_size != r_size)
+    {
+      size = min(l_size, r_size);
+      max_size = max(l_size, r_size);
+      diff_sizes = true;
+    }
+
+
+    // conduit zero initializes this array
+    conduit::Node res;
+    res["values"].set(conduit::DataType::float64(max_size));
+    double *res_array = res["values"].value();
+
+    Array<double> field_sums(res_array, max_size);
+
+    double *sums_ptr = field_sums.get_ptr(Exec::memory_space);
+
+    using for_policy = typename Exec::for_policy;
+
+    ascent::forall<for_policy>(0, size, [=] ASCENT_LAMBDA(index_t i)
+    {
+      const T val = l_accessor[i] + r_accessor[i];
+      sums_ptr[i] = val;
+    });
+    ASCENT_DEVICE_ERROR_CHECK();
+
+    if(diff_sizes)
+    {
+      if(l_size > r_size)
+      {
+        ascent::forall<for_policy>(size, l_size, [=] ASCENT_LAMBDA(index_t i)
+        {
+          const T val = l_accessor[i];
+          sums_ptr[i] = val;
+        });
+        ASCENT_DEVICE_ERROR_CHECK();
+      }
+      else
+      {
+        ascent::forall<for_policy>(size, r_size, [=] ASCENT_LAMBDA(index_t i)
+        {
+          const T val = r_accessor[i];
+          sums_ptr[i] = val;
+        });
+        ASCENT_DEVICE_ERROR_CHECK();
+      }
+    }
+
+    // synch the values back to the host
+    (void) field_sums.get_host_ptr();
+
```

Comment:
@cyrush My domain is going into this functor with "values", and is coming out with a "device_values" added onto it. 
ex:
    rho_electrons21:
      association: "element"
      topology: "topo"
      values: [0.0, 0.0, 0.0, ..., 0.0, 0.0]
      device_values: [0.0, 0.0, 0.0, ..., 0.0, 0.0]

Have you seen this before? Am I messing something up here?

I was able to add a field to nothing and got correct results. But then trying to add two fields together, it is as if it is only considering the second field. Not sure if it's pushing the input to "device_values" and then putting the fresh batch of values as "values"?

Reply:
Hmm, never mind. looks like device_values gets generated for the device calculations. So most likely unrelated as to why one field is overwriting the other.